File: SolvedProblems/SearchInABinarySearchTree/SearchInABinarySearchTree.js
const searchBST = (root, val) => {
  if (!root) return null;

  let queue = [root];
  let res = null;

  while (queue.length) {
    let curr = queue.pop();

    if (curr.val === val) {
      res = curr;
      break;
    }
    if (curr.left) queue.push(curr.left);
    if (curr.right) queue.push(curr.right);
  }

  return res;
};

File: SolvedProblems/SearchInABinarySearchTree/SearchInABinarySearchTree.py
def searchBST(root, val):
        queue = [root]
        res = None

        while (len(queue)):
            curr = queue.pop()

            if(curr.val == val):
                res = curr

            if(curr.left):
                queue.append(curr.left)
            if(curr.right):
                queue.append(curr.right)
        return res